megamek/src/megamek/common/MULParser.java

```diff
@@ -1183,6 +1161,15 @@ private void setPilotAttributes(Crew crew, int slot, Map<String,String> attribut
                 crew.setExternalIdAsString(attributes.get(EXT_ID), slot);
             }
 
+            if (attributes.containsKey(EXTRA_DATA)) {
+                Map<String, String> extraData = new HashMap<>();
+                String[] valuePairs = attributes.get(EXTRA_DATA).split(",");
+                for (int i = 0; i < valuePairs.length; i = i + 2) {
+                    //this is always in pairs, so it will not cause an out of bounds error
```

Comment:
As MUL's can come from anywhere, this isn't a safe assumption.

Reply:
Given how they MUST be saved in String, String pairs, anything else should break on load.
Because anything else means you've got invalid data saved.

Reply:
Then I'd rather it throw a specific exception than index out of range, but that's my druthers.